Scalar.Common/Platforms/Windows/CurrentUser.cs

```diff
@@ -153,6 +155,18 @@ public bool RunAs(string processName, string args)
                             try
                             {
                                 this.tracer.RelatedInfo("Started process '{0} {1}' with Id {2}", processName, args, procInfo.ProcessId);
+
+                                if (wait)
+                                {
+                                    SafeWait waiter = new SafeWait(procInfo.ProcessHandle);
+                                    waiter.WaitOne();
```

Comment:
I think we can use `WaitForSingleObject(procInfo.ProcessHandle);` and get rid of this wrapper class.  I haven't tested that though.

Reply:
That's much cleaner, thanks! I just ran my manual testing, so a commit is incoming.